### PR TITLE
Add Pushprox client and proxy

### PR DESCRIPTION
**Pull request by mcanevet**
#### Pull Request (PR) description
This PR allows to deploy pushprox client and proxy.

### CODE REVIEW

File: manifests/pushprox_client.pp [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+#  [*service_name*]
+#  Name of the pushprox_client service (default 'pushprox_client')
+#
+#  [*user*]
+#  User which runs the service
+#
+#  [*version*]
+#  The binary release version
+class prometheus::pushprox_client (
+  String $proxy_url,
```
Comment (bastelfreak) [inline manifests/pushprox_client.pp]: could you update the datatypes to prohibit empty strings?

File: manifests/pushprox_client.pp [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+  String $download_extension,
+  Variant[Stdlib::HTTPUrl, Stdlib::HTTPSUrl] $download_url_base,
+  Array $extra_groups,
```
Comment (bastelfreak) [inline manifests/pushprox_client.pp]: could you enforce the datatypes within the array?

File: manifests/pushprox_client.pp (hunk 1 above)
Comment (mcanevet) [reply manifests/pushprox_client.pp]: fixed

File: manifests/pushprox_client.pp (hunk 2 above)
Comment (mcanevet) [reply manifests/pushprox_client.pp]: fixed

File: manifests/pushprox_client.pp [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+#  [*service_name*]
+#  Name of the pushprox_client service (default 'pushprox_client')
+#
+#  [*user*]
+#  User which runs the service
+#
+#  [*version*]
+#  The binary release version
+class prometheus::pushprox_client (
+  String[1] $proxy_url,
```
Comment (bastelfreak) [inline manifests/pushprox_client.pp:78]: okay I thought you will add it for all the String types in both new classes :D

**Pull request merged**